x-pack/plugins/enterprise_search/server/lib/ml_inference_pipeline/get_inference_pipelines.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IngestPipeline, IngestProcessorContainer } from '@elastic/elasticsearch/lib/api/types';
+import { ElasticsearchClient } from '@kbn/core/server';
+import { MlTrainedModels } from '@kbn/ml-plugin/server';
+
+import { getMlModelConfigsForModelIds } from '../indices/fetch_ml_inference_pipeline_processors';
+
+/**
+ * Gets all ML inference pipelines. Redacts trained model IDs in those pipelines which reference
+ * a model inaccessible in the current Kibana space.
+ * @param esClient the Elasticsearch Client to use to fetch the errors.
+ * @param trainedModelsProvider ML trained models provider.
+ */
+export const getMlInferencePipelines = async (
+  esClient: ElasticsearchClient,
+  trainedModelsProvider: MlTrainedModels | undefined
+): Promise<Record<string, IngestPipeline>> => {
+  if (!trainedModelsProvider) {
+    return Promise.reject(new Error('Machine Learning is not enabled'));
+  }
+
+  // Fetch all ML inference pipelines
+  const fetchedInferencePipelines = await esClient.ingest.getPipeline({
+    id: 'ml-inference-*',
+  });
+
+  // Fetch all trained models from the current Kibana space, and identify those that have their
+  // model_id redacted
+  const trainedModels = await getMlModelConfigsForModelIds(esClient, trainedModelsProvider);
+  const redactedModelIds = Object.values(trainedModels)
+    .filter((modelConfig) => !modelConfig.modelId)
+    .map((modelConfig) => modelConfig.trainedModelName);
+
+  // Process pipelines: check if the model_id is one of the redacted ones, if so, redact it in the
+  // result as well
+  const inferencePipelinesResult: Record<string, IngestPipeline> = {};
+  Object.keys(fetchedInferencePipelines).forEach((name) => {
+    const inferencePipeline = fetchedInferencePipelines[name];
+
+    inferencePipelinesResult[name] = isModelIdRedacted(inferencePipeline, redactedModelIds)
```

Comment:
if `getInferenceProcessor` is not used outside of this module, and is only used in `isModelIdRedacted` and `redactModelId`, I'd recommend combining some items here... for example (not 100% sure this code is correct):
```
inferencePipelinesResult[name] = redactPipelineIfNecessary(inferencePipeline, redactedModelIds);
   ....
function redactPipelineIfNecessary(pipeline: IngestPipeline, redactedModelIds: string[]) : IngestPipeline {
  const pipelineCopy = { ...pipeline };
  const processor = getInferenceProcessor(pipelineCopy);
  if (!processor) {
    return IngestPipeline;
  }

  if (!processor?.inference || !redactedModelIds.includes(processor.inference.model_id)) {
    return ingestPipeline;
  }

  if (processor?.inference) {
    processor.inference.model_id = '';
  }
  return pipelineCopy;
}

```

Reply:
I ended up using a single convenience function - `redactModelIdIfInaccessible(processor: IngestProcessorContainer, accessibleModelIds: string[]): IngestProcessorContainer)`.